tools/grabber.js

```diff
@@ -118,21 +122,49 @@ var grabBlock = function (web3, blockHashOrNumber, listening) {
     }
 }
 
+var checkParentBlock = function (web3, blockData, recursively) {
+    if (blockData.number) {
+        var parentBlockNumber = blockData.number - 1;
+        Block.findOne({ number: parentBlockNumber }, function (err, b) {
+            if (err) {
+                console.log("Cannot find block in db:", err)
+            } else {
+                if (b) {
```

Comment:
else parent does not exist in the database? Shouldn't we fetch in that case as well?

Reply:
parent block is always there but added additional grabBlock call - everything could happens